Code/Editor/Plugins/EditorAssetImporter/SceneSerializationHandler.cpp

```diff
@@ -104,6 +98,38 @@ namespace AZ
         
         return scene;
     }
+    
+    bool SceneSerializationHandler::IsSceneCached(const AZStd::string& sceneFilePath)
+    {
+        if (!IsValidExtension(sceneFilePath))
+        {
+            return false;
+        }
+        AZ::IO::Path cleanPath(BuildCleanPathFromFilePath(sceneFilePath));
+        
+        CleanSceneMap();
+        // There's a small window where all shared pointers might be released after cleaning the map
+        // and before checking the list here, so this won't be 100% accurate, but it will still catch
+        // cases where the scene is in use somewhere.
+        auto sceneIt = m_scenes.find(cleanPath.Native());
+        if (sceneIt != m_scenes.end())
+        {
+            return true;
+        }
+        return false;
+
+    }
+
+    AZ::IO::Path SceneSerializationHandler::BuildCleanPathFromFilePath(const AZStd::string& filePath) const
+    {
+        AZ::IO::Path enginePath;
+        if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
```

Comment:
you don't need the nullptr check

```
 if (auto settingsRegistry = AZ::SettingsRegistry::Get())
```

Reply:
Fixed. I searched the original string because it was a copy paste, but there are over 60 results on it, so I don't want to divert that far from what I'm working on right now to clean up the rest.